src/frequenz/sdk/timeseries/_quantities.py

```diff
@@ -400,17 +413,41 @@ def as_megawatts(self) -> float:
         """
         return self._base_value / 1e6
 
+    @overload
+    def __mul__(self, percent: Percentage) -> Power:
+        """Return a power from multiplying this power by the given percentage.
+
+        Args:
+            percent: The percentage to multiply by.
+        """
+
+    @overload
     def __mul__(self, duration: timedelta) -> Energy:
         """Return an energy from multiplying this power by the given duration.
 
         Args:
             duration: The duration to multiply by.
+        """
 
-        Returns:
-            An energy from multiplying this power by the given duration.
+    def __mul__(self, other: Percentage | timedelta) -> Power | Energy:
+        """Return a power or energy from multiplying this power by the given value.
+
+        Args:
+            other: The percentage or duration to multiply by.
+
+        Raises:
+            TypeError: If the given value is not a percentage or duration.
         """
-        return Energy.from_watt_hours(
-            self._base_value * duration.total_seconds() / 3600.0
+        if isinstance(other, Percentage):
+            return super().__mul__(other)
+        if isinstance(other, timedelta):
+            return Energy.from_watt_hours(
+                self._base_value * other.total_seconds() / 3600.0
+            )
+
+        raise TypeError(
+            f"unsupported operand type(s) for *: '{self.__class__.__name__}' and "
+            f"'{other.__class__.__name__}'"
```

Comment:
I think this should raise a `NotImplementedError` to signal Python to either try other operators or print a nicer error message, right @shsms ?